bin/eslint-github-init.js

```diff
@@ -68,14 +72,59 @@ inquirer.prompt(questions).then(answers => {
     eslintrc.extends.push('plugin:github/browser')
   }
 
-  if (answers.flow) eslintrc.extends.push('plugin:github/flow')
+  if (answers.typeSystem === 'flow') eslintrc.extends.push('plugin:github/flow')
+  if (answers.typeSystem === 'typescript') {
+    eslintrc.extends.push('plugin:github/typescript')
+
+    // Create a `tsconfig.json` if one doesn't exits. Equivalent to running `tsc --init`.
+    const tsconfigPath = path.resolve(process.cwd(), 'tsconfig.json')
+    if (!fs.existsSync(tsconfigPath)) {
+      const tsconfigDefaults = {
+        compilerOptions: {
+          target: 'ES2015',
+          module: 'commonjs',
+          strict: true,
+          esModuleInterop: true
+        }
```

Comment:
We should set `"jsx": "react"` when user is using react.

Reply:
Also note that when we do that and read the `tsconfig.json` that it's a JSON5 file so we probably need to import a lib to parse it correctly.